tests/scripts/thread-cert/pktverify/consts.py

```diff
@@ -0,0 +1,137 @@
+#  1. Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+#  2. Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#  3. Neither the name of the copyright holder nor the
+#     names of its contributors may be used to endorse or promote products
+#     derived from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+from pktverify.bytes import Bytes
+from pktverify.ipv6_addr import Ipv6Addr
+
+DOMAIN_PREFIX = Bytes('fd00:7d03:7d03:7d03')
+BACKBONE_IPV6_PREFIX = Bytes('2001:0db8:0001:0000')
+
+# DOMAIN_PREFIX_REGEX_PATTERN = '^fd00:7d03:7d03:7d03:'
+# MESH_LOCAL_PREFIX = 'fdde:ad00:beef::/64'
+# MESH_LOCAL_PREFIX_REGEX_PATTERN = '^fdde:ad00:beef:(0){0,4}:'
+# ROUTING_LOCATOR = '64/:0:ff:fe00:/16'
+# ROUTING_LOCATOR_REGEX_PATTERN = r'.*:(0)?:0{0,2}ff:fe00:\w{1,4}$'
+# ALOC_FLAG_REGEX_PATTERN = '.*:fc..$'
+LINK_LOCAL_All_THREAD_NODES_MULTICAST_ADDRESS = Ipv6Addr('ff32:40:fdde:ad00:beef:0:0:1')
+REALM_LOCAL_All_THREAD_NODES_MULTICAST_ADDRESS = Ipv6Addr('ff33:40:fdde:ad00:beef:0:0:1')
+REALM_LOCAL_ALL_ROUTERS_ADDRESS = Ipv6Addr('ff03::2')
+LINK_LOCAL_ALL_NODES_MULTICAST_ADDRESS = Ipv6Addr('ff02::1')
+LINK_LOCAL_ALL_ROUTERS_MULTICAST_ADDRESS = Ipv6Addr('ff02::2')
+LINK_LOCAL_ALL_BBRS_MULTICAST_ADDRESS = Ipv6Addr('ff32:40:fd00:7d03:7d03:7d03:0:3')
+
+# MA in Test Plan, make sure these are same as ../config.py
+MA1 = Ipv6Addr('ff04::1234:777a:1')
+MA1g = Ipv6Addr('ff0e::1234:777a:1')
+MA2 = Ipv6Addr('ff05::1234:777a:1')
+MA3 = Ipv6Addr('ff0e::1234:777a:3')
+MA4 = Ipv6Addr('ff05::1234:777a:4')
+MA5 = Ipv6Addr('ff03::1234:777a:5')
+MA6 = Ipv6Addr('ff02::1')
+MAe1 = Ipv6Addr('fd0e::1234:777a:1')
+MAe2 = Ipv6Addr('::')
+MAe3 = Ipv6Addr('cafe::e0ff')
+# MAe4 = Ipv6Addr('ff05:0:0:0:1234:777a:0002')
+ALL_MPL_FORWARDERS_MA = Ipv6Addr('ff03::fc')
+
+LINK_LOCAL_PREFIX = Bytes("fe80")
+DEFAULT_MESH_LOCAL_PREFIX = Bytes("fd00:0db8:0000:0000")
+
+MLE_LINK_REQUEST = 0
```

Comment:
These are already defined in `mle.py`. Can we reuse it?

Reply:
I am aware that some constants in `pktverify` are already defined in other places. 
The reason I didn't use them directly is that I wanted `pktverify` to be a standalone module that does not depend on other modules. 

Currently `mle.py` is in the `thread-cert` directory, which is an upper layer of `pktverify`. 
`mle.py` also defines many MLE TLV factories and TLV classes, which is not required for `pktverify`. 

I think a better solution would be using a common module for Thread constants which is not available right now. 

Thoughts? @Irving-cl

Reply:
Sounds good to me.